irods/manager/data_object_manager.py

```diff
@@ -126,21 +130,23 @@ def put(self, local_path, irods_path, return_data_object = False, num_threads =
         else:
             obj = irods_path
 
-        with open(local_path, 'rb') as f, self.open(obj, 'w', **options) as o:
-
-            if self.should_parallelize_transfer (num_threads, f):
+        with open(local_path, 'rb') as f:
+            sizelist = []
+            if self.should_parallelize_transfer (num_threads, f, measured_obj_size = sizelist):
```

Comment:
That was a lesser-of-two-evils situation.  The logic would've been far more complicated without such a device, and Python offers no better choices for how to implement out-variables.  Would a comment be enough?

Reply:
java has a similar limitation.

does python have a class that makes it clear a parameter is an out-parameter?

Reply:
A search revealed these:

https://stackoverflow.com/questions/4702249/is-there-a-way-in-python-to-return-a-value-via-an-output-parameter
https://docs.python.org/3/faq/programming.html#how-do-i-write-a-function-with-output-parameters-call-by-reference

I don't believe there's anything along the lines of a decorator or annotation for this, and when you get down to it, and - simply put - the Python list is the best way to do it.  Some offer that returning tuples is a better way, but when the output parameter _may or may not_ be calcuated and returned, None becomes a possible second value of the tuple to be returned. In my view, this would harm the simple semantics of `should_parallelize...` returning a boolean, and muddle the logic of  the calling code even worse.

Reply:
Resolving for now.